engine/common/rpc/convert/execution_data.go

```diff
@@ -153,8 +153,8 @@ func TrieUpdateToMessage(t *ledger.TrieUpdate) (*entities.TrieUpdate, error) {
 	}
 
 	paths := make([][]byte, len(t.Paths))
-	for i, path := range t.Paths {
-		paths[i] = path[:]
+	for i := range t.Paths {
+		paths[i] = t.Paths[i][:]
```

Comment:
The two methods seem similar, and later changes to `t.Paths` seem to also impact `paths` (because array element pointers are equal) : https://go.dev/play/p/HD8XhfcFJDa
Maybe I didn't get what the bug is in the first place.

Reply:
the original bug was that the slice (and thus pointer to the underlying array) were reused resulting in the grpc response including a bunch of copies of the last element.

I opted to not make a copy here since there are likely to be a large number of paths, and the underlying data structure should be immutable.

Reply:
By updating slices to arrays, the issue and solution could be reproduced. Thank you!
https://go.dev/play/p/VMUv-aWUf8O